tests/parser/EclipseGridTests.cpp

```diff
@@ -2148,6 +2148,64 @@ BOOST_AUTO_TEST_CASE(TESTCP_ACTNUM_UPDATE) {
 }
 
 
+BOOST_AUTO_TEST_CASE(TESTCP_ACTNUM_AQUNUM) {
+    const char* deckData =
+
+        "RUNSPEC\n"
+        "\n"
+        "DIMENS\n"
+        " 3 2 1 /\n"
+        "GRID\n"
+        "COORD\n"
+        " 2000.0000  2000.0000  2000.0000   1999.9476  2000.0000  2002.9995\n"
+        " 2049.9924  2000.0000  2000.8726   2049.9400  2000.0000  2003.8722 \n"
+        " 2099.9848  2000.0000  2001.7452   2099.9324  2000.0000  2004.7448 \n"
+        " 2149.9772  2000.0000  2002.6179   2149.9248  2000.0000  2005.6174 \n"
+        " 2000.0000  2050.0000  2000.0000   1999.9476  2050.0000  2002.9995 \n"
+        " 2049.9924  2050.0000  2000.8726   2049.9400  2050.0000  2003.8722 \n"
+        " 2099.9848  2050.0000  2001.7452   2099.9324  2050.0000  2004.7448 \n"
+        " 2149.9772  2050.0000  2002.6179   2149.9248  2050.0000  2005.6174 \n"
+        " 2000.0000  2100.0000  2000.0000   1999.9476  2100.0000  2002.9995 \n"
+        " 2049.9924  2100.0000  2000.8726   2049.9400  2100.0000  2003.8722 \n"
+        " 2099.9848  2100.0000  2001.7452   2099.9324  2100.0000  2004.7448 \n"
+        " 2149.9772  2100.0000  2002.6179   2149.9248  2100.0000  2005.6174 / \n"
+        "ZCORN\n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2000.0000  2000.8726  2000.8726  2001.7452  2001.7452  2002.6179 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 \n"
+        " 2002.9995  2003.8722  2003.8722  2004.7448  2004.7448  2005.6174 / \n"
+        "ACTNUM\n"
+        " 0 0 1 1 0 1 / \n"
+        "PORO\n"
+        "  6*0.15 /\n"
+        "AQUNUM\n"
+        "    1     1  1  1   1000000.0   10000   0.25   400    2585.00   285.00    1   1  /\n"
+        "    1     2  1  1   1000000.0   10000   0.25   400    2585.00   285.00    1   1  /\n"
+        "/\n"
+        "AQUCON\n"
+        "   1     3    3      2    2     1   1   'J+'      1.00      1  /\n"
+        "/\n"
+        "EDIT\n"
+        "\n";
+
+    Opm::Parser parser;
+    const auto deck = parser.parseString( deckData) ;
+    Opm::EclipseGrid grid( deck);
+
+    const std::vector<int>& grid_actnum = grid.getACTNUM();
+    const std::vector<int> desired_actnum = {1, 1, 1, 1, 0, 1};
+
+    BOOST_CHECK( grid_actnum.size() == desired_actnum.size() );
```

Comment:
If you are lazy you can compare vectors with just `BOOST_CHECK( v1 == v2 )`. The disadvantage of this is that it will not print a nice comprehensible message if the check fails.